orderer/consensus/smartbft/util_network_test.go

```diff
@@ -404,14 +427,64 @@ func createBFTChainUsingMocks(t *testing.T, node *Node) (*smartbft.BFTChain, err
 		return egressCommMock
 	}
 
-	localConfigCluster := localconfig.Cluster{ReplicationPolicy: "simple"}
+	synchronizerMock := smartBFTMocks.NewSynchronizer(t)
+	synchronizerMock.EXPECT().Sync().RunAndReturn(
+		func() types.SyncResponse {
+			t.Logf("Sync Called by node %v", node.NodeId)
+			// iterate over the ledger of the other nodes and find the highest ledger to sync with
+			max := 0
+			var ledgerToCopy []*cb.Block
+			for _, n := range node.nodesMap {
+				if n.NodeId != node.NodeId {
+					ledgerToCopy = n.State.GetLedgerArray()
+					len := len(ledgerToCopy)
+					if len > max {
+						max = len
+					}
+				}
+			}
+
+			// sync node
+			for i := node.State.GetLedgerHeight(); i < len(ledgerToCopy); i++ {
+				clonedBlock := proto.Clone(ledgerToCopy[i]).(*cb.Block)
+				node.State.AddBlock(clonedBlock)
+			}
+
+			// send response
+			// at this point the chain exists, so we can use its methods
+			return types.SyncResponse{
+				Latest: *node.Chain.BlockToDecision(ledgerToCopy[len(ledgerToCopy)-1]),
+				Reconfig: types.ReconfigSync{
+					InReplicatedDecisions: false,
+					CurrentNodes:          []uint64{1, 2, 3, 4},
```

Comment:
this should be derived from the network definition no?

Reply:
This is true. I already changed it in the next PR, but I will fix it in the current PR to be accurate.